Add docs for proper kustomization installs

mac-chaffee opened the pull request:
Closes #840 

This PR adds documentation for installing awx-operator natively with Kustomize, rather than requireing the user to clone the repo. This way, users can provide their own patches and additional resources on top of the base resources.

I also regenerated the table of contents and added a comment telling users how to regenerate it like #793 found.

mac-chaffee committed:
Add docs for proper kustomization installs

shanemcd commented on README.md:
```diff
@@ -101,46 +106,66 @@ Let's create an alias for easier usage:
+kind: Kustomization
+resources:
+  # Find the latest tag here: https://github.com/ansible/awx-operator/releases
+  - github.com/ansible/awx-operator/config/default?ref=0.18.0
```
Please remove the hardcoded versions and include a placeholder. This is already outdated and we've done a lot of work to avoid the need for bumping versions in files before doing a release.

shanemcd reviewed (changes requested):
Hello. Thanks for this, it's a great contribution. One small request, otherwise this all looks awesome.

mac-chaffee committed:
Remove explicit tags to avoid need to bump versions

Signed-off-by: Mac Chaffee <machaffe@renci.org>

kurokobo commented:
Thanks for PR, I have two comments.

> `kustomize build . | kubectl apply -f -`

This requires that the `kustomize` command have to be installed, but there is no instruction to install `kustomize`.
The same things can be achieved by `kubectl appky -k .` without `kustomize` command, in my opinion this is much easier than installing/using `kustomize` command. What do you think?

> Next, create a file named `awx-demo.yml` in the same folder
> ...
> ```yaml
>   # Add this extra line:
>   - awx-demo.yaml
> ```

Two filenames are not matched (`yaml`/`yml`).

mac-chaffee committed:
Mention how to install Kustomize.

It's recommended to install the standalone version of kustomize rather
than using the version that ships with kubectl because that version is
typically very old and doesn't match the docs.

mac-chaffee commented:
The reason I didn't recommend using the version of kustomize that comes with kubectl is because that version is typically quite outdated. Looks like the situation is improving since [kubectl v1.23 is catching up](https://github.com/kubernetes/kubectl/blob/release-1.23/go.mod#L44), but as recently as [kubectl v1.20](https://github.com/kubernetes/kubectl/blob/7873ea42e5df4038630d4251620cdb9978d132c7/go.mod#L47) is two major versions behind.

I find that the one thing harder than keeping my clusters up-to-date is making sure all my users have updated clients ;)

shanemcd reviewed (approved):
Awesome contribution. Thanks a ton.
